packages/go_router/lib/src/delegate.dart

```diff
@@ -73,25 +79,36 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
 
   /// Returns `true` if the active Navigator can pop.
   bool canPop() {
-    final _NavigatorStateIterator iterator = _createNavigatorStateIterator();
```

Comment:
previously we need to scan the list and use parent navigator key to jump around the list. Therefore i created this iterator to abstract away the logic.

After the refactor we just need to look at the last node in the RouteMatchList and walk up the tree branch.